### ISSUE TITLE
insmod: ERROR: could not insert module

### ISSUE THREAD

**Issue by ruurdadema** [bondagit/aes67-linux-daemon#76]
When running `run_test.sh` I get back some errors:

```
~/aes67-linux-daemon$ sudo ./run_test.sh S24_3LE 48000 2 5
Compiling tools ...
make: Nothing to be done for 'all'.
Creating test file ...
test file created
Creating configuration files ..
net.ipv4.igmp_max_memberships = 66
insmod: ERROR: could not insert module 3rdparty/ravenna-alsa-lkm/driver/MergingRavennaALSA.ko: Unknown symbol in module
Starting PTP master ...
Starting AES67 daemon ...
Waiting for PTP slave to sync ...
[2022-03-31 23:01:10.343728] [0x00007f268cd76480] [fatal]   driver_handler:: init open: Protocol not supported
[2022-03-31 23:01:10.343752] [0x00007f268cd76480] [fatal]   Kernel module not loaded ?
[2022-03-31 23:01:10.343804] [0x00007f268cd76480] [fatal]   main:: fatal exception error: DriverManager:: init failed
daemon exiting with code: 1
```

The problem seems to originate by a failure to install the kernel module:

```
~/aes67-linux-daemon$ sudo insmod 3rdparty/ravenna-alsa-lkm/driver/MergingRavennaALSA.ko
insmod: ERROR: could not insert module 3rdparty/ravenna-alsa-lkm/driver/MergingRavennaALSA.ko: Unknown symbol in module
```

Any idea what might be causing this or how to debug this?

Running on Ubuntu 20.04.4 LTS.

**Comment by ruurdadema** [comment]
Running `dmesg | tail` give me this:

```
~/aes67-linux-daemon$ dmesg | tail
[ 1343.998275] MergingRavennaALSA: Unknown symbol snd_card_new (err -2)
[ 1343.998287] MergingRavennaALSA: Unknown symbol snd_interval_list (err -2)
[ 1343.998296] MergingRavennaALSA: Unknown symbol snd_pcm_lib_ioctl (err -2)
[ 1343.998308] MergingRavennaALSA: Unknown symbol snd_pcm_lib_get_vmalloc_page (err -2)
[ 1343.998317] MergingRavennaALSA: Unknown symbol snd_ctl_notify (err -2)
[ 1343.998326] MergingRavennaALSA: Unknown symbol snd_pcm_set_ops (err -2)
[ 1343.998335] MergingRavennaALSA: Unknown symbol snd_pcm_hw_constraint_list (err -2)
[ 1343.998368] MergingRavennaALSA: Unknown symbol snd_pcm_period_elapsed (err -2)
[ 1343.998376] MergingRavennaALSA: Unknown symbol snd_pcm_hw_constraint_step (err -2)
[ 1343.998389] MergingRavennaALSA: Unknown symbol snd_pcm_format_width (err -2)
```

**Comment by ruurdadema** [comment]
For future self:

I realised I was running on a system without an actual audio device, which caused ALSO to be not loaded. After attaching an external USB audio device (and rebooting_ the ALSA symbols were found.

After this change I ran into a different error: `module_layout version incompatibility`, which was caused by the fact that the driver was built without the ALSO module enabled. Rebuilding fixed the issue.
